cpp/src/arrow/type.h

```diff
@@ -2126,7 +2126,7 @@ Result<std::shared_ptr<Schema>> UnifySchemas(
 
 namespace internal {
 
-static inline bool HasValidityBitmap(Type::type id) {
+inline constexpr bool HasValidityBitmap(Type::type id) {
```

Comment:
This was mentioned in another PR of mine, but I'm pushing it here instead to reduce the noise there.

Reply:
constexpr functions are implicitly inline so we don't need the inline keywork here, right?

Reply:
Pushed a change.

Reply:
I think the arrow codebase tends to omit inline keyword for constexpr functions, for example in type_trait.h and [this PR](https://github.com/apache/arrow/pull/14241). 
P.S. I don't think the inline specifier has much to do with inline function expansion for modern compilers. It's more for making sure ODR is not violated.

Reply:
Yes, according to the C++ reference, "A constexpr specifier used in a function declaration implies inline".
https://en.cppreference.com/w/cpp/language/constexpr